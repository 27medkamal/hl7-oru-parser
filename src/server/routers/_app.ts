import { createCallerFactory, publicProcedure, router } from '../trpc';
import { analyseRoute } from './anaylse';

export const appRouter = router({
  healthcheck: publicProcedure.query(() => 'yay!'),
  analyse: analyseRoute,
});

export const createCaller = createCallerFactory(appRouter);
export type AppRouter = typeof appRouter;
