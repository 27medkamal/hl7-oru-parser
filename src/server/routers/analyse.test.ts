import { createContextInner } from '../context';
import { createCaller } from './_app';
import * as fs from 'fs';

test('Analyses file snapshot', async () => {
  const ctx = await createContextInner({});
  const caller = createCaller(ctx);

  expect(
    await caller.analyse({
      oruFileContent: fs
        .readFileSync('prisma/data/MP826520.oru.txt')
        .toString(),
    }),
  ).toMatchSnapshot();
});
