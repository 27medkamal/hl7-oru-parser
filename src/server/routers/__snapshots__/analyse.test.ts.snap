// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Analyses file snapshot 1`] = `
{
  "data": {
    "CRP": {
      "C-reactive Protein": [
        {
          "diagnosticName": "C-reactive Protein",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "< 3.33",
          "groupName": "CRP",
          "groupNames": undefined,
          "name": "C-reactive Protein",
          "result": "2",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "< 5",
          "units": "mg/L",
        },
      ],
    },
    "ESR": {
      "Erythrocyte Sedimentation Rate": [
        {
          "diagnosticName": "Erythrocyte Sedimentation Rate",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "0.91 - 20",
          "groupName": "ESR",
          "groupNames": undefined,
          "name": "Erythrocyte Sedimentation Rate",
          "result": "15",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "1 - 20",
          "units": "mm/hr",
        },
      ],
    },
    "EUC": {
      "Electrolyte Panel": [
        {
          "diagnosticName": "Electrolyte Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "122.73 - 161.11",
          "groupName": "EUC",
          "groupNames": undefined,
          "name": "Serum Sodium",
          "result": "145",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "135 - 145",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Electrolyte Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "3.18 - 6.11",
          "groupName": "EUC",
          "groupNames": undefined,
          "name": "Serum Potassium",
          "result": "4.4",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "3.5 - 5.5",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Electrolyte Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "86.36 - 122.22",
          "groupName": "EUC",
          "groupNames": undefined,
          "name": "Serum Chloride",
          "result": "105",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "95 - 110",
          "units": "mmol/L",
        },
      ],
      "Kidney Function": [
        {
          "diagnosticName": "Kidney Function",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "2.73 - 8.33",
          "groupName": "EUC",
          "groupNames": undefined,
          "name": "Serum Urea",
          "result": "5.3",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "3 - 7.5",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Kidney Function",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "54.55 - 122.22",
          "groupName": "EUC",
          "groupNames": undefined,
          "name": "Serum Creatinine",
          "result": "71",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "60 - 110",
          "units": "umol/L",
        },
      ],
    },
    "FBE": {
      "Haemoglobin and Red Cell Analysis": [
        {
          "diagnosticName": "Haemoglobin and Red Cell Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "0.35 - 0.6",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Haematocrit",
          "result": "0.4",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "0.39 - 0.54",
          "units": "",
        },
        {
          "diagnosticName": "Haemoglobin and Red Cell Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "3.43 - 5.7",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Red Cell Count",
          "result": "4.4",
          "standardAtRisk": {
            "classification": "Yes",
            "text": "Red Cell Count and Morphology out of Normal Reference Range",
          },
          "standardRange": "4.5 - 6.5",
          "units": "x10*12/L",
        },
        {
          "diagnosticName": "Haemoglobin and Red Cell Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "72.73 - 111.11",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Mean Corpuscular Volume (MCV)",
          "result": "91",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "80 - 100",
          "units": "fL",
        },
        {
          "diagnosticName": "Haemoglobin and Red Cell Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "24.55 - 35.56",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Mean Corpuscular Haemoglobin (MCH)",
          "result": "31",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "27 - 32",
          "units": "pg",
        },
        {
          "diagnosticName": "Haemoglobin and Red Cell Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "281.82 - 400",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Mean Corpuscular Haemoglobin Concentration (MCHC)",
          "result": "337",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "310 - 360",
          "units": "g/L",
        },
        {
          "diagnosticName": "Haemoglobin and Red Cell Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "9.09 - 16.67",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Red Cell Distribution Width (RDW)",
          "result": "13.8",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "10 - 15",
          "units": "",
        },
      ],
      "Platelet Count": [
        {
          "diagnosticName": "Platelet Count",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "136.36 - 500",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Platelets",
          "result": "211",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "150 - 450",
          "units": "x10*9/L",
        },
      ],
      "White Cell Count and Profile": [
        {
          "diagnosticName": "White Cell Count and Profile",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "1.82 - 8.33",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Neutrophils",
          "result": "2.1",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "2 - 7.5",
          "units": "x10*9/L",
        },
        {
          "diagnosticName": "White Cell Count and Profile",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "0.91 - 4.44",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Lymphocytes",
          "result": "1.7",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "1 - 4",
          "units": "x10*9/L",
        },
        {
          "diagnosticName": "White Cell Count and Profile",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "< 1.11",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Monocytes",
          "result": "0.3",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "< 1",
          "units": "x10*9/L",
        },
        {
          "diagnosticName": "White Cell Count and Profile",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "< 0.56",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Eosinophils",
          "result": "0.1",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "< 0.5",
          "units": "x10*9/L",
        },
        {
          "diagnosticName": "White Cell Count and Profile",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "< 0.33",
          "groupName": "FBE",
          "groupNames": undefined,
          "name": "Basophils",
          "result": "0",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "< 0.3",
          "units": "x10*9/L",
        },
      ],
    },
    "FPU uACR": {
      "Urinary Albumin-to-Creatinine Ratio (Early Kidney Protein Loss Detection)": [
        {
          "diagnosticName": "Urinary Albumin-to-Creatinine Ratio (Early Kidney Protein Loss Detection)",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "",
          "groupName": "FPU uACR",
          "groupNames": undefined,
          "name": "Random Urinary Albumin",
          "result": "< 3",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "",
          "units": "mg/L",
        },
        {
          "diagnosticName": "Urinary Albumin-to-Creatinine Ratio (Early Kidney Protein Loss Detection)",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "",
          "groupName": "FPU uACR",
          "groupNames": undefined,
          "name": "Random Urinary Creatinine",
          "result": "7.8",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Urinary Albumin-to-Creatinine Ratio (Early Kidney Protein Loss Detection)",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "< 2.78",
          "groupName": "FPU uACR",
          "groupNames": undefined,
          "name": "Urinary Albumin-to-Creatinine Ratio",
          "result": "< 0.4",
          "standardAtRisk": {
            "classification": "Yes",
            "text": "Possible Kidney Impairment",
          },
          "standardRange": "> 3.5",
          "units": "mg/mmol",
        },
      ],
    },
    "Fasting BSL": {
      "Fasting Glucose": [
        {
          "diagnosticName": "Fasting Glucose",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "3.27 - 5.89",
          "groupName": "Fasting BSL",
          "groupNames": undefined,
          "name": "Fasting Glucose",
          "result": "5.3",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "3.6 - 6",
          "units": "mmol/L",
        },
      ],
    },
    "Folate": {
      "Serum Folate": [
        {
          "diagnosticName": "Serum Folate",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "> 5.45",
          "groupName": "Folate",
          "groupNames": undefined,
          "name": "Serum Folate",
          "result": "28",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "> 6",
          "units": "nmol/L",
        },
      ],
    },
    "HBA1c": {
      "HbA1c Test": [
        {
          "diagnosticName": "HbA1c Test",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "3.64 - 6",
          "groupName": "HBA1c",
          "groupNames": undefined,
          "name": "HbA1c (NGSP)",
          "result": "5.3",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "4 - 5.6",
          "units": "%",
        },
        {
          "diagnosticName": "HbA1c Test",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "18.18 - 40",
          "groupName": "HBA1c",
          "groupNames": undefined,
          "name": "HbA1c (IFCC)",
          "result": "34",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "20 - 38",
          "units": "mmol/mol",
        },
        {
          "diagnosticName": "HbA1c Test",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "",
          "groupName": "HBA1c",
          "groupNames": undefined,
          "name": "Estimated Average Glucose",
          "result": "5.9",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "",
          "units": "mmol/L",
        },
      ],
    },
    "Iron Studies": {
      "Iron Levels Analysis": [
        {
          "diagnosticName": "Iron Levels Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "4.55 - 33.33",
          "groupName": "Iron Studies",
          "groupNames": undefined,
          "name": "Serum Iron",
          "result": "8",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "5 - 30",
          "units": "umol/L",
        },
        {
          "diagnosticName": "Iron Levels Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "1.82 - 3.56",
          "groupName": "Iron Studies",
          "groupNames": undefined,
          "name": "Serum Transferrin",
          "result": "2.3",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "2 - 3.2",
          "units": "g/L",
        },
        {
          "diagnosticName": "Iron Levels Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "9.09 - 50",
          "groupName": "Iron Studies",
          "groupNames": undefined,
          "name": "Transferrin Saturation",
          "result": "14",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "10 - 45",
          "units": "%",
        },
        {
          "diagnosticName": "Iron Levels Analysis",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "27 - 444.44",
          "groupName": "Iron Studies",
          "groupNames": undefined,
          "name": "Serum Ferritin",
          "result": "135",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "30 - 500",
          "units": "ng/mL",
        },
      ],
    },
    "LFTs": {
      "Liver Function Panel": [
        {
          "diagnosticName": "Liver Function Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "3.64 - 22.22",
          "groupName": "LFTs",
          "groupNames": undefined,
          "name": "Serum Bilirubin",
          "result": "6",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "4 - 20",
          "units": "umol/L",
        },
        {
          "diagnosticName": "Liver Function Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "31.82 - 122.22",
          "groupName": "LFTs",
          "groupNames": undefined,
          "name": "Serum Alkaline Phosphatase",
          "result": "53",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "35 - 110",
          "units": "U/L",
        },
        {
          "diagnosticName": "Liver Function Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "4.55 - 55.56",
          "groupName": "LFTs",
          "groupNames": undefined,
          "name": "Serum Gamma-GT",
          "result": "12",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "5 - 50",
          "units": "U/L",
        },
        {
          "diagnosticName": "Liver Function Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "4.55 - 44.44",
          "groupName": "LFTs",
          "groupNames": undefined,
          "name": "Serum Alanine Aminotransferase",
          "result": "17",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "5 - 40",
          "units": "U/L",
        },
        {
          "diagnosticName": "Liver Function Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "9.09 - 44.44",
          "groupName": "LFTs",
          "groupNames": undefined,
          "name": "Serum Aspartate Transaminase",
          "result": "19",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "10 - 40",
          "units": "U/L",
        },
        {
          "diagnosticName": "Liver Function Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "61.82 - 94.44",
          "groupName": "LFTs",
          "groupNames": undefined,
          "name": "Serum Total Protein",
          "result": "76",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "68 - 85",
          "units": "g/L",
        },
        {
          "diagnosticName": "Liver Function Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "35.45 - 55.56",
          "groupName": "LFTs",
          "groupNames": undefined,
          "name": "Serum Albumin",
          "result": "45",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "39 - 50",
          "units": "g/L",
        },
        {
          "diagnosticName": "Liver Function Panel",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "20.91 - 43.33",
          "groupName": "LFTs",
          "groupNames": undefined,
          "name": "Serum Globulin",
          "result": "31",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "23 - 39",
          "units": "g/L",
        },
      ],
    },
    "Lipids/HDL/LDL": {
      "Full Cholesterol Profiling": [
        {
          "diagnosticName": "Full Cholesterol Profiling",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "3.18 - 6.11",
          "groupName": "Lipids/HDL/LDL",
          "groupNames": undefined,
          "name": "Total Cholesterol",
          "result": "4.4",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "3.5 - 5.5",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Full Cholesterol Profiling",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "< 1.24",
          "groupName": "Lipids/HDL/LDL",
          "groupNames": undefined,
          "name": "Serum Triglycerides",
          "result": "0.5",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "< 1.5",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Full Cholesterol Profiling",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "1.09 - 2.9",
          "groupName": "Lipids/HDL/LDL",
          "groupNames": undefined,
          "name": "HDL Cholesterol",
          "result": "1.32",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "> 1",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Full Cholesterol Profiling",
          "everlabAtRisk": {
            "classification": "Yes",
            "text": "Increased Atherosclerotic Disease Risk Due to Suboptimal Lipid Profile",
          },
          "everlabRange": "< 2.89",
          "groupName": "Lipids/HDL/LDL",
          "groupNames": undefined,
          "name": "LDL Cholesterol",
          "result": "2.9",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "< 3.5",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Full Cholesterol Profiling",
          "everlabAtRisk": {
            "classification": "Yes",
            "text": "Suboptimal Lipid Profile for Arterial Health",
          },
          "everlabRange": "< 2.78",
          "groupName": "Lipids/HDL/LDL",
          "groupNames": undefined,
          "name": "Non HDL Cholesterol",
          "result": "3.1",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "< 3.9",
          "units": "mmol/L",
        },
      ],
    },
    "Mg": {
      "Serum Magnesium": [
        {
          "diagnosticName": "Serum Magnesium",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "0.68 - 1.11",
          "groupName": "Mg",
          "groupNames": undefined,
          "name": "Serum Magnesium",
          "result": "0.94",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "0.65 - 1",
          "units": "mmol/L",
        },
      ],
    },
    "Serum Calcium": {
      "Serum Calcium": [
        {
          "diagnosticName": "Serum Calcium",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "1.95 - 2.83",
          "groupName": "Serum Calcium",
          "groupNames": undefined,
          "name": "Serum Calcium",
          "result": "2.3",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "2.15 - 2.55",
          "units": "mmol/L",
        },
        {
          "diagnosticName": "Serum Calcium",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "1.95 - 2.83",
          "groupName": "Serum Calcium",
          "groupNames": undefined,
          "name": "Corrected Calcium",
          "result": "2.2",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "2.15 - 2.55",
          "units": "mmol/L",
        },
      ],
    },
    "Serum Phosphate": {
      "Serum Phosphate": [
        {
          "diagnosticName": "Serum Phosphate",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "0.73 - 1.67",
          "groupName": "Serum Phosphate",
          "groupNames": undefined,
          "name": "Serum Phosphate",
          "result": "1.29",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "0.8 - 1.5",
          "units": "mmol/L",
        },
      ],
    },
    "TSH": {
      "Thyroid Stimulating Hormone (TSH)": [
        {
          "diagnosticName": "Thyroid Stimulating Hormone (TSH)",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "0.45 - 4.44",
          "groupName": "TSH",
          "groupNames": undefined,
          "name": "Thyroid Stimulating Hormone (TSH)",
          "result": "1.75",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "0.5 - 5.5",
          "units": "mIU/L",
        },
      ],
    },
    "Vit B12": {
      "Vitamin B12": [
        {
          "diagnosticName": "Vitamin B12",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "181.82 - 777.78",
          "groupName": "Vit B12",
          "groupNames": undefined,
          "name": "Vitamin B12",
          "result": "423",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "200 - 700",
          "units": "pmol/L",
        },
      ],
    },
    "Vit D": {
      "Vitamin D3": [
        {
          "diagnosticName": "Vitamin D3",
          "everlabAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "everlabRange": "45.45 - 277.78",
          "groupName": "Vit D",
          "groupNames": undefined,
          "name": "Vitamin D3",
          "result": "52",
          "standardAtRisk": {
            "classification": "No",
            "text": "No risk",
          },
          "standardRange": "50 - 250",
          "units": "nmol/L",
        },
      ],
    },
  },
  "personalDetails": {
    "age": 55,
    "firstName": "JOHN",
    "gender": "Male",
    "lastName": "SMITH",
  },
}
`;
